### PR TITLE
Examples of quantization with accuracy control

### PR DESCRIPTION
**Pull request by aleksandr-mokrov**
ticket: CVS-111190
This PR provides two notebooks with examples of quantization with accuracy control.

### CODE REVIEW

File: notebooks/122-quantizing-model-with-accuracy-control/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Quantization with accuracy control using NNCF
+
+This tutorial demonstrates how to apply `INT8` quantization with accuracy control to the speech recognition model and to PyTorch YOLOv8 model.
```
Comment (AlexKoff88) [inline notebooks/122-quantizing-model-with-accuracy-control/README.md:3]: I think it is two tutorials. So, it is better to rephrase it like `These tutorials demonstrate how to apply 8-bit quantization with accuracy control to the speech recognition and YOLOv8 models:` and then continue with the list.
I also wonder why it is two-in-one at the same time.

File: notebooks/122-quantizing-model-with-accuracy-control/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+Supported models:
+* `122-speech-recognition-quantization-wav2vec2.ipynb` demonstrates how to apply post-training `INT8` quantization with accuracy control on a fine-tuned [Wav2Vec2-Base-960h](https://huggingface.co/facebook/wav2vec2-base-960h) [PyTorch](https://pytorch.org/) model, trained on the [LibriSpeech ASR corpus](https://www.openslr.org/12).
+* `122-yolov8-quantization-with-accuracy-control.ipynb` demonstrates how to apply post-training `INT8` quantization on the [YOLOv8](https://github.com/ultralytics/) PyTorch model
+
+The code of the tutorials is designed to be extendable to custom models and datasets.
```
Comment (AlexKoff88) [inline notebooks/122-quantizing-model-with-accuracy-control/README.md:9]: Really? Sounds too generic. I would remove it or rephrase it to `extendable to the same model types trained on custom datasets`.

File: notebooks/122-quantizing-model-with-accuracy-control/README.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+The advanced quantization flow allows to apply 8-bit quantization to the model with control of accuracy metric. This is achieved by keeping the most impactful operations within the model in the original precision. The flow is based on the [Basic 8-bit quantization](https://docs.openvino.ai/2023.0/basic_quantization_flow.html) and has the following differences:
```
Comment (AlexKoff88) [inline notebooks/122-quantizing-model-with-accuracy-control/README.md:11]: I believe it is based on https://docs.openvino.ai/2023.0/quantization_w_accuracy_control.html. Please fix.

File: notebooks/122-quantizing-model-with-accuracy-control/122-speech-recognition-quantization-wav2vec2.ipynb [hunk 4]
```diff
@@ -0,0 +1,712 @@
...
+    "ov_model = openvino.convert_model(torch_model, example_input=default_input)"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "<a id=\"4\"></a>\n",
```
Comment (AlexKoff88) [inline notebooks/122-quantizing-model-with-accuracy-control/122-speech-recognition-quantization-wav2vec2.ipynb:187]: I am really concerned about creating own objects to work with datasets and metrics while everything is already available in public. I don't think that these how people go about this when they train/validate models.
LibriSpeech is available both in [torchaudio](https://pytorch.org/audio/main/generated/torchaudio.datasets.LIBRISPEECH.html) and HuggingFace [datasets](https://huggingface.co/datasets/librispeech_asr) library. Word Error rate is available in [torchmetrics](https://pytorch.org/torcheval/main/generated/torcheval.metrics.WordErrorRate.html) as well. 
You are doing a role modeling here for the users. I don't think they will be happy to write everything from scratch just to optimize the model.

File: notebooks/122-quantizing-model-with-accuracy-control/README.md (hunk 1 above)
Comment (aleksandr-mokrov) [reply notebooks/122-quantizing-model-with-accuracy-control/README.md]: Rephrased

File: notebooks/122-quantizing-model-with-accuracy-control/README.md (hunk 2 above)
Comment (aleksandr-mokrov) [reply notebooks/122-quantizing-model-with-accuracy-control/README.md]: Rephrased

File: notebooks/122-quantizing-model-with-accuracy-control/README.md (hunk 3 above)
Comment (aleksandr-mokrov) [reply notebooks/122-quantizing-model-with-accuracy-control/README.md]: Fixed

File: notebooks/122-quantizing-model-with-accuracy-control/122-speech-recognition-quantization-wav2vec2.ipynb (hunk 4 above)
Comment (aleksandr-mokrov) [reply notebooks/122-quantizing-model-with-accuracy-control/122-speech-recognition-quantization-wav2vec2.ipynb]: Redesigned. Now it is used a lightweight huggingface dataset `librispeech_asr_dummy` with preprocessing function, WordErrorRate from torchmetrics, Wav2Vec2Processor for pre- and post-processing steps and no direct file downloads.

File: notebooks/122-quantizing-model-with-accuracy-control/122-yolov8-quantization-with-accuracy-control.ipynb [hunk 5]
```diff
@@ -0,0 +1,339 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "source": [
+    "# Convert and Optimize YOLOv8 with OpenVINO™\n",
+    "\n",
+    "The YOLOv8 algorithm developed by Ultralytics is a cutting-edge, state-of-the-art (SOTA) model that is designed to be fast, accurate, and easy to use, making it an excellent choice for a wide range of object detection, image segmentation, and image classification tasks.\n",
```
[the remark this answers is not in the document]
Comment (aleksandr-mokrov) [reply notebooks/122-quantizing-model-with-accuracy-control/122-yolov8-quantization-with-accuracy-control.ipynb:8]: Details were removed

File: notebooks/122-quantizing-model-with-accuracy-control/122-yolov8-quantization-with-accuracy-control.ipynb [hunk 6]
```diff
@@ -0,0 +1,332 @@
+{
```
Comment (sbalandi) [inline notebooks/122-quantizing-model-with-accuracy-control/122-yolov8-quantization-with-accuracy-control.ipynb:1]: <p>Could you add more information about ValidatorClass and purpose of data loader?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1234/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (sbalandi) [inline notebooks/122-quantizing-model-with-accuracy-control/122-yolov8-quantization-with-accuracy-control.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #7.    validator = model.ValidatorClass(args)
</pre><p>duplicate of str 1</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1234/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (sbalandi) [inline notebooks/122-quantizing-model-with-accuracy-control/122-yolov8-quantization-with-accuracy-control.ipynb:1]: <p>looks like it makes sense to compare the original and quantized model,&nbsp;to show what become better (size maybe) and to show that the accuracy has not drop</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1234/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (sbalandi) [inline notebooks/122-quantizing-model-with-accuracy-control/122-yolov8-quantization-with-accuracy-control.ipynb:1]: <p>I got error <code>TypeError: cannot pickle 'Model' object </code>with nncf version git+https://github.com/openvinotoolkit/nncf.git@develop. nncf==2.5.0 work slower, but without fail. Do you know, maybe it is required additional modules to install ?</p><p> </p><p><code>File c:\users\sbalandi\me\work\1\openvino_env\lib\site-packages\nncf\quantization\algorithms\accuracy_control\ranker.py:213, in Ranker._multiprocessing_calculation_ranking_score(self, quantized_model, quantized_model_graph, groups_to_rank, ranking_subset_indices, reference_values_for_each_item)</code></p><p><code>&nbsp;&nbsp;208 for idx, current_group in enumerate(groups_to_rank):</code></p><p><code>&nbsp;&nbsp;209&nbsp;&nbsp;&nbsp;modified_model = revert_operations_to_floating_point_precision(</code></p><p><code>&nbsp;&nbsp;210&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;current_group.operations, current_group.quantizers, quantized_model, quantized_model_graph</code></p><p><code>&nbsp;&nbsp;211&nbsp;&nbsp;&nbsp;)</code></p><p><code>--&gt; 213&nbsp;&nbsp;&nbsp;prepared_model_queue.append(self._algo_backend.prepare_for_inference_async(modified_model))</code></p><p><code>&nbsp;&nbsp;215&nbsp;&nbsp;&nbsp;if idx &gt;= (self._num_processes - 1):</code></p><p><code>&nbsp;&nbsp;216&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;prepared_model = prepared_model_queue.pop(0).get()</code></p><p><code>File c:\users\sbalandi\me\work\1\openvino_env\lib\site-packages\nncf\quantization\algorithms\accuracy_control\openvino_backend.py:123, in OVAccuracyControlAlgoBackend.prepare_for_inference_async(model)</code></p><p><code>&nbsp;&nbsp;121 queue = multiprocessing.Queue()</code></p><p><code>&nbsp;&nbsp;122 p = multiprocessing.Process(target=compile_model, args=(model, queue))</code></p><p><code>--&gt; 123 p.start()</code></p><p><code>&nbsp;&nbsp;124 return OVAsyncPreparedModel(p, queue)</code></p><p><code>File ~\AppData\Local\Programs\Python\Python38\lib\multiprocessing\process.py:121, in BaseProcess.start(self)</code></p><p><code>&nbsp;&nbsp;118 assert not _current_process._config.get('daemon'), \</code></p><p><code>&nbsp;&nbsp;119&nbsp;&nbsp;&nbsp;&nbsp;'daemonic processes are not allowed to have children'</code></p><p><code>&nbsp;&nbsp;120 _cleanup()</code></p><p><code>--&gt; 121 self._popen = self._Popen(self)</code></p><p><code>&nbsp;&nbsp;122 self._sentinel = self._popen.sentinel</code></p><p><code>&nbsp;&nbsp;123 # Avoid a refcycle if the target function holds an indirect</code></p><p><code>&nbsp;&nbsp;124 # reference to the process object (see bpo-30775)</code></p><p><code>File ~\AppData\Local\Programs\Python\Python38\lib\multiprocessing\context.py:224, in Process._Popen(process_obj)</code></p><p><code>&nbsp;&nbsp;222 @staticmethod</code></p><p><code>&nbsp;&nbsp;223 def _Popen(process_obj):</code></p><p><code>--&gt; 224&nbsp;&nbsp;&nbsp;return _default_context.get_context().Process._Popen(process_obj)</code></p><p><code>File ~\AppData\Local\Programs\Python\Python38\lib\multiprocessing\context.py:327, in SpawnProcess._Popen(process_obj)</code></p><p><code>&nbsp;&nbsp;324 @staticmethod</code></p><p><code>&nbsp;&nbsp;325 def _Popen(process_obj):</code></p><p><code>&nbsp;&nbsp;326&nbsp;&nbsp;&nbsp;from .popen_spawn_win32 import Popen</code></p><p><code>--&gt; 327&nbsp;&nbsp;&nbsp;return Popen(process_obj)</code></p><p><code>File ~\AppData\Local\Programs\Python\Python38\lib\multiprocessing\popen_spawn_win32.py:93, in Popen.__init__(self, process_obj)</code></p><p><code>&nbsp;&nbsp;&nbsp;91 try:</code></p><p><code>&nbsp;&nbsp;&nbsp;92&nbsp;&nbsp;&nbsp;reduction.dump(prep_data, to_child)</code></p><p><code>---&gt; 93&nbsp;&nbsp;&nbsp;reduction.dump(process_obj, to_child)</code></p><p><code>&nbsp;&nbsp;&nbsp;94 finally:</code></p><p><code>&nbsp;&nbsp;&nbsp;95&nbsp;&nbsp;&nbsp;set_spawning_popen(None)</code></p><p><code>File ~\AppData\Local\Programs\Python\Python38\lib\multiprocessing\reduction.py:60, in dump(obj, file, protocol)</code></p><p><code>&nbsp;&nbsp;&nbsp;58 def dump(obj, file, protocol=None):</code></p><p><code>&nbsp;&nbsp;&nbsp;59&nbsp;&nbsp;&nbsp;'''Replacement for pickle.dump() using ForkingPickler.'''</code></p><p><code>---&gt; 60&nbsp;&nbsp;&nbsp;ForkingPickler(file, protocol).dump(obj)</code></p><p><code>TypeError: cannot pickle 'Model' object</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1234/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

File: notebooks/122-quantizing-model-with-accuracy-control/122-speech-recognition-quantization-wav2vec2.ipynb (hunk 4 above)
Comment (AlexKoff88) [reply notebooks/122-quantizing-model-with-accuracy-control/122-speech-recognition-quantization-wav2vec2.ipynb]: Thanks, now it makes sense.

**Review by AlexKoff88 — APPROVED** [review]

File: notebooks/122-quantizing-model-with-accuracy-control/122-yolov8-quantization-with-accuracy-control.ipynb (hunk 6 above)
Comment (aleksandr-mokrov) [reply notebooks/122-quantizing-model-with-accuracy-control/122-yolov8-quantization-with-accuracy-control.ipynb:1]: Added

Comment (aleksandr-mokrov) [reply notebooks/122-quantizing-model-with-accuracy-control/122-yolov8-quantization-with-accuracy-control.ipynb:1]: Removed

Comment (aleksandr-mokrov) [reply notebooks/122-quantizing-model-with-accuracy-control/122-yolov8-quantization-with-accuracy-control.ipynb:1]: I tried to reproduce it in a new environment with python3.8 and python3.10  and ubuntu 22.04.  There are no errors. Looks like a bug in nncf in a specific environment.

Comment (aleksandr-mokrov) [reply notebooks/122-quantizing-model-with-accuracy-control/122-yolov8-quantization-with-accuracy-control.ipynb:1]: Accuracy and performance comparison were added.

Comment (sbalandi) [reply notebooks/122-quantizing-model-with-accuracy-control/122-yolov8-quantization-with-accuracy-control.ipynb:1]: <p>sorry, I don't see, where ?</p>

Comment (sbalandi) [reply notebooks/122-quantizing-model-with-accuracy-control/122-yolov8-quantization-with-accuracy-control.ipynb:1]: @aleksandr-mokrov yes, you are right, it is bug - CVS-118321 on win, work around for windows now is adding parameter `num_ranking_processes=1` to advanced_accuracy_restorer_parameters (https://github.com/openvinotoolkit/nncf/blob/develop/nncf/quantization/advanced_parameters.py#L201) . Could you please add it ?

Comment (aleksandr-mokrov) [reply notebooks/122-quantizing-model-with-accuracy-control/122-yolov8-quantization-with-accuracy-control.ipynb:1]: Something went wrong.  I've updated, please check again.

File: notebooks/122-quantizing-model-with-accuracy-control/122-speech-recognition-quantization-wav2vec2.ipynb [hunk 7]
```diff
@@ -0,0 +1,457 @@
+{
```
Comment (eaidova) [inline notebooks/122-quantizing-model-with-accuracy-control/122-speech-recognition-quantization-wav2vec2.ipynb:1]: <p> &gt;<strong>NOTE</strong>: Currently, 8-bit quantization with accuracy control is available only for models in OpenVINO representation. </p><p> </p><p>sounds like you plan other use cases, maybe need to add that it is related to NNCF?</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1234/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (eaidova) [inline notebooks/122-quantizing-model-with-accuracy-control/122-speech-recognition-quantization-wav2vec2.ipynb:1]: <p>header is misleading, because you compare accuracy, not performance.</p><p>And also probably, it will be useful if you compare FP16 model too, to see that there is no degradation after model conversion</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1234/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

File: notebooks/122-quantizing-model-with-accuracy-control/122-yolov8-quantization-with-accuracy-control.ipynb (hunk 6 above)
Comment (aleksandr-mokrov) [reply notebooks/122-quantizing-model-with-accuracy-control/122-yolov8-quantization-with-accuracy-control.ipynb:1]: Added

File: notebooks/122-quantizing-model-with-accuracy-control/122-speech-recognition-quantization-wav2vec2.ipynb (hunk 7 above)
Comment (aleksandr-mokrov) [reply notebooks/122-quantizing-model-with-accuracy-control/122-speech-recognition-quantization-wav2vec2.ipynb:1]: Added

**Review by eaidova — APPROVED** [review]

**Review by sbalandi — APPROVED** [review]

**Review by as-suvorov — APPROVED** [review]
